core/src/main/kotlin/com/expediagroup/sdk/core/client/BaseRapidClient.kt

```diff
@@ -39,7 +41,15 @@ abstract class BaseRapidClient(
             clientConfiguration.toProvider(),
             RapidConfigurationProvider
         )
-    private val _httpClient: HttpClient = buildHttpClient(_configurationProvider, AuthenticationStrategy.AuthenticationType.SIGNATURE, httpClientEngine)
+
+    private val _httpClientEngine: HttpClientEngine = clientConfiguration.okHttpClient?.let {
```

Comment:
The underscore '_' prefix is reserved for backing properties with a corresponding public getter. For example:

```
private val _configurationProvider: ConfigurationProvider = ...  // backing property
val configurationProvider: ConfigurationProvider                 // public getter
    get() = _configurationProvider
```

Since this property doesn't have a public getter, please rename it to `httpClientEngine`.